esp-hal/src/uart.rs

```diff
@@ -1862,41 +1954,22 @@ impl UartRx<'_, Async> {
                 events |= RxEvent::FifoTout;
             }
 
-            let events_happened = UartRxFuture::new(self.uart.reborrow(), events).await;
-            // always drain the fifo, if an error has occurred the data is lost
-            let read = self.flush_buffer(buf);
-            // check error events
-            rx_event_check_for_error(events_happened)?;
-            // Unfortunately, the uart's rx-timeout counter counts up whenever there is
-            // data in the fifo, even if the interrupt is disabled and the status bit
-            // cleared. Since we do not drain the fifo in the interrupt handler, we need to
-            // reset the counter here, after draining the fifo.
-            self.regs()
```

Comment:
Looks like this work around has gone completely, are you sure we don't still need it?

Reply:
Yes. The hardware fires the FIFO full interrupt, we handle it, return data, etc., and then the hardware at some point fires the timeout. We don't handle the timeout here, but we may in a subsequent read. In that case, if there is no data in the fifo, we just detect that the fifo is empty, and loop back to waiting for another interrupt.